engine/src/vs_logging.cpp

```diff
@@ -117,4 +102,77 @@ void VegaStrikeLogger::FlushLogs() {
     fflush(stderr);
 }
 
+VegaStrikeLogger::VegaStrikeLogger() {
+    logging_core_ = boost::log::core::get();
+    boost::log::add_common_attributes();
+    slg_ = boost::make_shared<boost::log::sources::severity_logger_mt<vega_log_level>>();
+
+    console_log_sink_ = boost::log::add_console_log
+            (
+                    std::cerr,
+                    boost::log::keywords::format =
+                            "%Message%",                                                    /*< log record format specific to the console >*/
+                    boost::log::keywords::auto_flush =
+                            true /*false*/                                                  /*< whether to do the equivalent of fflush(stdout) after every msg >*/
+            );
+    file_log_sink_ = nullptr;
+}
+
+VegaStrikeLogger::~VegaStrikeLogger() {
+    FlushLogs();
+}
+
+void VegaStrikeLogger::Log(const vega_log_level level, const std::string &message) {
+    boost::log::record rec = slg_->open_record(boost::log::keywords::severity = level);
+    if (rec)
+    {
+        boost::log::record_ostream strm(rec);
+        strm << message;
+        strm.flush();
+        slg_->push_record(boost::move(rec));
+    }
+}
+
+void VegaStrikeLogger::LogAndFlush(const vega_log_level level, const std::string &message) {
+    Log(level, message);
+    FlushLogs();
+}
+
+void VegaStrikeLogger::Log(const vega_log_level level, const char *message) {
+    boost::log::record rec = slg_->open_record(boost::log::keywords::severity = level);
+    if (rec)
+    {
+        boost::log::record_ostream strm(rec);
+        strm << message;
+        strm.flush();
+        slg_->push_record(boost::move(rec));
+    }
```

Comment:
nice - so much better than us handling it directly.

Reply:
This is taken from the Boost.Log documentation, incidentally. I basically copied the code for one of their macros, albeit with one or two adjustments.